### PR TITLE
Add on prem troubleshooting guides

### PR DESCRIPTION
**Pull request by vverbani**
New page in the debugging series on troubleshooting guides for a tyk-self managed instance.

### CODE REVIEW

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+tags: ["Debugging", "On-Prem Debugging", "Debugging Guide", "Debugging Tips" ]
+description: "Debugging Tips and Tricks on how-to debug a Tyk Instance"
+menu:
+  main:
+    parent: "Debugging Series"
+aliases:
+  - /debugging-series/on-prem-debugging/
+---
+
+Debugging our on-prem environment is a challenging task by any means. Many moving pieces could be causing the error you see, i.e. gateway, dashboard, backend service, Mongo, Redis, Postgres, pump, load balancer(s), certificates etc.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: If this is in the docs I would recommend removing the opinion based language and leave things more neutral.

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+You will eventually hit the point of error by further isolating parts of your API.
+
+## What do you do when you can’t fix the error?
+You're probably not the first to encounter this error. Visit these relevant Tyk resources additional help or guidance:
+
+1. [Tyk Community Forums](https://community.tyk.io/)
+2. [Tyk FAQ Section]({{< ref "frequently-asked-questions" >}})
+3. [Tyk Github Page](https://github.com/TykTechnologies)
+4. Tyk Support (Paid SLA only)
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: no link

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+You should be doing two things from the beginning: isolating your components to see where the error is coming from and enabling debug logs to ensure you get all the information you need.
+
+## `/Hello` endpoint
+
+The `/hello` endpoints are the quickest way to get a feel for if the overall environment is up and running.
+
+You can call the same endpoint against both the gateway and the dashboard. The dashboard returns much less info but is still valuable regarding isolating issues.
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```json
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+menu:
+  main:
+    parent: "Debugging Series"
+aliases:
+  - /debugging-series/on-prem-debugging/
+---
+
+Debugging our on-prem environment is a challenging task by any means. Many moving pieces could be causing the error you see, i.e. gateway, dashboard, backend service, Mongo, Redis, Postgres, pump, load balancer(s), certificates etc.
+
+You should be doing two things from the beginning: isolating your components to see where the error is coming from and enabling debug logs to ensure you get all the information you need.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: This is a nice starting point but language-wise would be better written as something like 

This guide should help a user of Tyks self managed pro offering in debugging common issues. A helpful way to go about this is by:
1.  Isolating your components to see where the error is coming 
2. enabling debug logs to ensure you get all the information you need

Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Also not clear to me what exactly isolating components means at this stage in the doc

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+## `/Hello` endpoint
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: lower case

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+The `/hello` endpoints are the quickest way to get a feel for if the overall environment is up and running.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: endpoints? Is this a consistent tool across the stack, are there exceptions i.e. you can rewrite the name of this in gateway

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+You can call the same endpoint against both the gateway and the dashboard. The dashboard returns much less info but is still valuable regarding isolating issues.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Better still to say what information we expect from a call to each component and why we present it.

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+            componentType: "datatstore",
+            time: "2022-11-18T13:30:33Z"
+        }
+    }
+}
+```
+
+## Gateway
+
+This gives you a clearer picture of whether the above components are working. While it doesn't give you the whole picture, it certainly narrows the field of vision that much more.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Opinionated language needs cleaning up.

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+## Debug Logs
+
+Having debug logs turned on helps close the gap in which component is throwing the error.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Same again

Comment (joshblakeley) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: Also not really anything concrete here. Why?

**Review by letzya — CHANGES_REQUESTED** [review]
reminder to get peer review

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+### Gateway Debug Settings
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
+"log_level": "debug"
+```
+
+If you’re using a `*.env` for your configuration:
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Address the k8s/helm case here too?

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 2 above)
Comment (vverbani) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: I left the Tyk support blank on purpose. Atm we get a bunch of spam tickets from non-SLA clients. SLA clients should already have this link during their signing/onboarding process

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+```
+TYK_DB_LOGLEVEL=debug
+```
+
+You can find the full log levels in our documentation.
+
+## Versions/ Replicate
+
+If you are using an older version of Tyk and running into an error, the issue might resolve if you upgrade.
```
Comment (joshblakeley) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: What do we recommend here more explicitly?

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 3 above)
Comment (vverbani) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: this was left without `json` on purpose. The json file should be in the `{"key": "value"}` format but in the above api return it's `{key: "value"}` and this gets flagged with bunch of red in the hugo html UI process.

**Review by joshblakeley — DISMISSED** [review]
Few notes to cover off. Only got halfway through but will revisit. Thanks for the effort so far - this will be awesome when published!

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 1 above)
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Lets align with our stance on the nomenclature, could reword the Tyk components as

Tyk Gateway, Tyk Dashboard, Tyk Pump,

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+You can read version release notes and upgrade your Tyk versions here: [Version Releases](https://github.com/TykTechnologies/tyk/releases) .
+
+## Dashboard
+
+It's essential to note that our dashboard is built on top of our OSS, which means that the dashboard uses the Dashboard API to retrieve its data. This means you can use the developer tools on your browser to access the API and its information. From here, you'll be able to see whether it's the Dashboard UI that's causing the error or if it's the API itself.
```
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Lets firm up our reference to the dashboard as Tyk Dashboard

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 3 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: Are you sure? not in my gw
image

**Review by letzya — CHANGES_REQUESTED** [review]

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 11 above)
Comment (sedkis) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: Add "Please see the latest version releases [here](release pages).  We recommend you to use the latest version of Tyk always.  Tyk is backwards compatible, upgrading to newer versions won't activate any features or change behaviour of your existing environment"

**Comment by vverbani** [comment]
@joshblakeley Want to review the changes

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+If you're using Tyk Helm Charts. Add the following items to your `values.yaml`:
+
+```
+extraEnvs:
+  - name: TYK_LOGLEVEL
+    value: debug
+```
+
+You can find the full log levels in our documentation.
```
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: It would be helpful if we linked to an example or what TYK_LOGLEVEL represents here

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+This guide should help a user of Tyks self managed pro offering in debugging common issues. A helpful way to go about this is by:
+
+1. Isolating your components to see where the error is coming
+2. Enabling debug logs to ensure you get all the information you need
+
+## Gateway `/hello` endpoint
+
+The `/hello` health check endpoint against the Gateway is the quickest way in determining the status of your Tyk instance. You can find more information in our docs about the [Gateway Liveness health check]({{< ref "planning-for-production/ensure-high-availability/health-check" >}}).
+
+The reason this endpoint is important is that components such as the Tyk Dashboard, RPC, or Redis can go down at any given point so it is useful to know if the Gateway is currently usable or not.
```
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Lets adjust our language here, we don't want to say that the components can go down at any point because they're unlikely to- perhaps we could reword this to:

This endpoint is important as it allows for a user to isolate where a problem might be originating from. At a glance, the `/hello` endpoint reports the Tyk-Gateway's connectivity to the Tyk-Dashboard, Redis, or in enterprise/hybrid installations: MDCB (rpc).

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+        },
+        "rpc": {
+            "status": "pass",
+            "componentType": "system",
+            "time":"2023-01-13T14:45:00Z"
+        }
+}
+```
+
+The Gateway can still function if the Dashboard fails. This means that your API calls are still working as normal.
```
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: (Perhaps we could also expand a bit about what will work and what will not work if each component fails to connect?)
If the Tyk-Dashboard or RPC connectivity fails, the Tyk-Gateway will still function based on the last recieved configurations from those components respectively. If Redis fails, Tyk-Gateway will not be able to perform since it is a hard dependency.

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+## Debug Logs
+
+Having Debug mode turned on is helpful when debugging because it throws more descriptive error messages.
```
Comment (JRWu) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Perhaps having an example here of what the different log levels represent would be useful, i.e.

INFO: *some log example here*

DEBUG: *some log example here*

**Review by JRWu — DISMISSED** [review]
Suggesting some rephrasing & content additions

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+tags: ["Debugging", "On-Prem Debugging", "Debugging Guide", "Debugging Tips" ]
+description: "Debugging Tips and Tricks on how-to debug a Tyk Instance"
+menu:
+  main:
+    parent: "Debugging Series"
+aliases:
+  - /debugging-series/on-prem-debugging/
+---
+
+This guide should help a user of Tyks self managed pro offering in debugging common issues. A helpful way to go about this is by:
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
This guide should help a user of Tyk's self managed offering in debugging common issues. A helpful way to go about this is by:
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+
+1. Isolating your components to see where the error is coming
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
1. Isolating your components to see where the error is coming from
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+2. Enabling debug logs to ensure you get all the information you need
+
+## Gateway `/hello` endpoint
+
+The `/hello` health check endpoint against the Gateway is the quickest way in determining the status of your Tyk instance. You can find more information in our docs about the [Gateway Liveness health check]({{< ref "planning-for-production/ensure-high-availability/health-check" >}}).
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
Querying the gateway's `/hello` health endpoint is the quickest way to determine the status of your Tyk instance. You can find more information in our docs about the [Gateway Liveness health check]({{< ref "planning-for-production/ensure-high-availability/health-check" >}}).
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+
+This endpoint is important as it allows for a user to isolate where a problem might be originating from. At a glance, the `/hello` endpoint reports the Gateways connectivity to the Dashboard, Redis, or in enterprise/hybrid installations: MDCB (rpc).
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
This endpoint is important as it allows the user to isolate the problem's origin. At a glance, the `/hello` endpoint reports the Gateways connectivity to Redis, and the control plane components eg. Dashboard, MDCB, Tyk Cloud. 
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+
+```json
+{
+    "status":"pass",
+    "version":"v4.0.10",
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Maybe have the latest version when you release the docs. It shows that we are actively working on our docs 🙃

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+        },
+        "rpc": {
+            "status": "pass",
+            "componentType": "system",
+            "time":"2023-01-13T14:45:00Z"
+        }
+}
+```
+
+If the Dashboard or RPC connectivity fails, the Gateway will still function based on the last received configurations from those components respectively. If Redis fails, Gateway will not be able to perform since it is a hard dependency.
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
If the Dashboard or RPC connectivity fails (control plane components), the Gateway will still function based on the last received configurations from those components. However, if Redis fails, Gateway will go down since it is a hard dependency.
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 23]
```diff
@@ -0,0 +1,189 @@
...
+
+## Debug Logs
+
+Having Debug mode turned on is helpful when debugging because it throws more descriptive error messages. For example, here are the different outputs you receive when calling an Open Keyless API with `info` and `debug` log level modes.
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
Setting the log level to debug will allow for more descriptive logs that will give a better context around any issue you might be facing. For example, here are the different outputs you receive when calling an Open Keyless API with `info` and `debug` log level modes.
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+tyk-pump       | time="Jan 24 14:32:20" level=info msg="Purged 1 records..." prefix=mongo-pump-selective
+tyk-pump       | time="Jan 24 14:32:20" level=info msg="Completed purging the records" collection="tyk_analytics" number of records=1 prefix=mongo-pump
+tyk-pump       | time="Jan 24 14:32:20" level=info msg="Purged 1 records..." prefix=mongo-pump
+tyk-mongo      | 2023-01-24T14:32:20.398+0000 I  NETWORK  [listener] connection accepted from 172.20.0.3:54712 #19 (19 connections now open)
+tyk-pump       | time="Jan 24 14:32:20" level=info msg="Completed upserting" collection="tyk_analytics_aggregates" prefix=mongo-pump-aggregate
+tyk-pump       | time="Jan 24 14:32:20" level=info msg="Purged 1 records..." prefix=mongo-pump-aggregate
+
+```
+
+As shown above, the `debug` log level mode throws more information which will help during your debugging stage, i.e when the API call was started, when it was finished, how long it took for the call to finish, the endpoint that was called, the upstream that was called, the organization that the API belongs to, and more.
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
As shown above, the `debug` log level mode provides more information which will help during your debugging stage, i.e when the API call was started, when it was finished, how long it took for the call to finish, the endpoint that was called, the upstream that was called, the organization that the API belongs to, and more.
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+
+### Gateway Debug Settings
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
+"log_level": "debug"
+```
+
+If you’re using a `*.env` for your configuration:
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
If you’re using environment variables for your configuration:
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+
+### Dashboard Debug Settings
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
+"log_level": "debug"
+```
+
+If you’re using a `*.env` for your configuration:
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
If you’re using environment variables for your configuration:
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+
+```
+extraEnvs:
+  - name: TYK_LOGLEVEL
+    value: debug
+```
+
+You can find the full [log levels]({{< ref "log-data" >}}) in our documentation.
+
+## Versions/ Replicate
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: Why Replicate?

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+
+Please see the latest version releases [here](https://github.com/TykTechnologies/tyk/releases).
+
+We recommend using the latest version of Tyk always. Tyk is backwards compatible, upgrading to newer versions won't activate any features or change behaviour of your existing environment
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: I would say latest LTS not latest, no?

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 29]
```diff
@@ -0,0 +1,189 @@
...
+
+## Dashboard
+
+It's essential to note that our Dashboard is built on top of our OSS, which means that the Dashboard uses the Dashboard API to retrieve its data. This means you can use the developer tools on your browser to access the API and its information. From here, you'll be able to see whether it's the Dashboard UI that's causing the error or if it's the API itself.
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: This does not make sense.

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 30]
```diff
@@ -0,0 +1,189 @@
...
+
+By using the browser developer tools, you can see and replicate this information through API calls.
+
+### Isolating
+
+As mentioned above, errors can come from many parts of Tyk or your architecture, and as such, one of the critical things you'll pick up during your debugging phase is isolating these environments.
+
+### Dashboard Level
+
+When using the Dashboard, you'll run into a common theme where something doesn't work on the Dashboard but will work when using the Tyk Gateway API.
```
Comment (zalbiraw) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]: I do think we should say "its a common theme"

**Review by zalbiraw — DISMISSED** [review]

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 31]
```diff
@@ -0,0 +1,189 @@
...
+tyk-pump       | time="Jan 24 14:39:19" level=info msg="Purged 1 records..." prefix=mongo-pump
+tyk-pump       | time="Jan 24 14:39:19" level=info msg="Purged 1 records..." prefix=mongo-pump-selective
+tyk-mongo      | 2023-01-24T14:39:19.228+0000 I  NETWORK  [listener] connection accepted from 172.20.0.2:51028 #19 (19 connections now open)
+tyk-pump       | time="Jan 24 14:39:19" level=info msg="Completed upserting" collection="tyk_analytics_aggregates" prefix=mongo-pump-aggregate
+tyk-pump       | time="Jan 24 14:39:19" level=info msg="Purged 1 records..." prefix=mongo-pump-aggregate
+```
+
+Here is a more detailed output of the same call when using `debug` as the log level:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```bash
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 32]
```diff
@@ -0,0 +1,189 @@
...
+
+If the Dashboard or RPC connectivity fails, the Gateway will still function based on the last received configurations from those components respectively. If Redis fails, Gateway will not be able to perform since it is a hard dependency.
+
+## Debug Logs
+
+Having Debug mode turned on is helpful when debugging because it throws more descriptive error messages. For example, here are the different outputs you receive when calling an Open Keyless API with `info` and `debug` log level modes.
+
+Here is the output when using `info` as the log level:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```bash
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 33]
```diff
@@ -0,0 +1,189 @@
...
+
+As shown above, the `debug` log level mode throws more information which will help during your debugging stage, i.e when the API call was started, when it was finished, how long it took for the call to finish, the endpoint that was called, the upstream that was called, the organization that the API belongs to, and more.
+
+### Gateway Debug Settings
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```json
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 34]
```diff
@@ -0,0 +1,189 @@
...
+"log_level": "debug"
+```
+
+If you’re using a `*.env` for your configuration:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```json
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 35]
```diff
@@ -0,0 +1,189 @@
...
+TYK_GW_LOGLEVEL=debug
+```
+
+If you're using Tyk Helm Charts. Add the following items to your `values.yaml`:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```yaml
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 36]
```diff
@@ -0,0 +1,189 @@
...
+
+If you’re using environment variables for your configuration:
+
+```
+TYK_DB_LOGLEVEL=debug
+```
+
+If you're using Tyk Helm Charts. Add the following items to your `values.yaml`:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```yaml
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 37]
```diff
@@ -0,0 +1,189 @@
...
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
+"log_level": "debug"
+```
+
+If you’re using environment variables for your configuration:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```yaml
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 38]
```diff
@@ -0,0 +1,189 @@
...
+extraEnvs:
+  - name: TYK_LOGLEVEL
+    value: debug
+```
+
+### Dashboard Debug Settings
+
+If you’re using a `*.conf` for your configuration parameters:
+
+```
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```json
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 39]
```diff
@@ -0,0 +1,189 @@
...
+## Gateway `/hello` endpoint
+
+Querying the gateway's `/hello` health endpoint is the quickest way to determine the status of your Tyk instance. You can find more information in our docs about the [Gateway Liveness health check]({{< ref "planning-for-production/ensure-high-availability/health-check" >}}).
+
+This endpoint is important as it allows the user to isolate the problem's origin. At a glance, the `/hello` endpoint reports the Gateways connectivity to Redis, and the control plane components eg. Dashboard, MDCB, Tyk Cloud. 
+
+```json
+{
+    "status":"pass",
+    "version":"v4.0.10",
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
    "version":"v5.0",
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 21 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: Changed to 5.0 thinking we'll push it next week

**Comment by letzya** [comment]
@vverbani Hi, what do you need in order to get this done? Thanks

**Comment by letzya** [comment]
@vverbani Hi, checking again, what do you need in order to get this done? happy to help

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 29 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: fixing now

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 28 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: fixing

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 27 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: fixing

File: tyk-docs/content/debugging-series/on-prem-debugging.md (hunk 30 above)
Comment (letzya) [reply tyk-docs/content/debugging-series/on-prem-debugging.md]: fixing

**Comment by letzya** [comment]
This PR https://github.com/TykTechnologies/tyk-docs/pull/3492 - addresses all the comments from @zalbiraw  - Please review it so we can publish this pr

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 40]
```diff
@@ -0,0 +1,191 @@
...
+As a next step to this investigation, if that specific endpoint exists also in [Tyk Gateway API]({{< ref "tyk-gateway-api" >}}), you can compare the responses from both gateway and dashboard requests. 
+
+### Isolating
+
+As mentioned above, errors can happen in any of the components of your Tyk deployment, and as such, one of the critical things you'll pick up during your debugging phase is isolating these environments.
+
+### Dashboard Level
+
+When using the Dashboard, you'll run into a common theme where something doesn't work on the Dashboard but will work when using the Tyk Gateway API.
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 41]
```diff
@@ -0,0 +1,190 @@
...
+
+The Dashboard front-end (GUI included) uses [Tyk Dashboard API]({{< ref "tyk-dashboard-api" >}}) to retrieve data to display or update. This means you can use the [developer tools on your browser](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Tools_and_setup/What_are_browser_developer_tools) to access the API and its information. Looking into the API details, the URL, the headers, the payload and the response can help you investigate the source of the issue and replicate it with API calls using an HTTP client such as [cURL](https://curl.se/) or [Postman](https://www.postman.com/).
+As a next step to this investigation, if that specific endpoint exists also in [Tyk Gateway API]({{< ref "tyk-gateway-api" >}}), you can compare the responses from both gateway and dashboard requests. 
+
+### Isolating
+
+As mentioned above, errors can happen in any of the components of your Tyk deployment, and as such, one of the critical things you'll pick up during your debugging phase is isolating these environments.
+
+### Dashboard Level
+
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
```

File: tyk-docs/content/debugging-series/on-prem-debugging.md [hunk 42]
```diff
@@ -0,0 +1,189 @@
...
+When debugging an issue, in order to isolate the gateway from the Dashboard, try to do the same API call on both Tyk Dashboard and Tyk Gateway
```
Comment (letzya) [inline tyk-docs/content/debugging-series/on-prem-debugging.md]:
```suggestion
When debugging an issue, in order to isolate the gateway from the Dashboard, try to call the same API ednpoint on both Tyk Dashboard and Tyk Gateway 
```

**Review by letzya — APPROVED** [review]
